File: app/src/main/java/com/keelim/arducon/SettingFragment.java
package com.keelim.arducon;

import android.os.Bundle;
import android.support.v7.preference.ListPreference;
import android.support.v7.preference.PreferenceFragmentCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

public class SettingFragment extends PreferenceFragmentCompat {

    private ListPreference mListPreference;


    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        // Indicate here the XML resource you created above that holds the preferences
        setPreferencesFromResource(R.xml.settings_preference, rootKey);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        mListPreference = (ListPreference) getPreferenceManager().findPreference("preference_key");


        return inflater.inflate(R.layout.main_fragment, container, false);
    }
}

File: app/src/main/java/com/keelim/arducon/view/setting/SettingFragment.kt
package com.keelim.arducon.view.setting

import android.os.Bundle
import android.widget.Toast
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import com.keelim.arducon.R

class SettingFragment : PreferenceFragmentCompat() {
    //setting
    override fun onCreatePreferences(savedInstanceState: Bundle, rootKey: String) {
        addPreferencesFromResource(R.xml.setting_preference)
    }

    override fun onPreferenceTreeClick(preference: Preference): Boolean {
        val key = preference.key
        when(key){
            "developer" -> Toast.makeText(context, "개발 진행 중입니다.", Toast.LENGTH_SHORT).show()

        }
        return super.onPreferenceTreeClick(preference)
    }


}